src/t_zset.c

```diff
@@ -511,6 +541,21 @@ unsigned long zslGetRank(zskiplist *zsl, double score, sds ele) {
     return 0;
 }
 
+/* Find the rank for an element by both score and key.
+ * Returns 0 when the element cannot be found, rank otherwise.
+ * Note that the rank is 1-based due to the span of zsl->header to the
+ * first element. */
+unsigned long zslGetRankByNode(zskiplist *zsl, zskiplistNode *x) {
+    int i = zslGetNodeHeight(x);
+    unsigned long rank = zslGetNodeSpanAtHeight(x, i);
+    while (x->level[zslGetNodeHeight(x)].forward) {
```

Comment:
The height is the top level index? If level is 0-based, I would think the top level is `height - 1`.

Maybe we should either change this to `zslGetNodeHeight(x) - 1` or rename height to top-level as `zslGetNodeTopLevelIndex(x)`?

Reply:
Yes I keep the height as the top level index. I think it might be more readable if I keep the height as the number of levels.

Reply:
done